pick.lic

```diff
@@ -2,523 +2,665 @@
-    @pet_count = 0
-
-    @settings = get_settings(args.flex)
-
-    if args.source
-      source = args.source.split('=')[1]
-      @settings.picking_box_source = source
+  def do_buffs
+    if @settings.lockpick_buff_bot && DRRoom.pcs.include?(@settings.lockpick_buff_bot)
+      DRC.bput("whisper #{@settings.lockpick_buff_bot} buff hol", "You whisper")
     end
 
-    if @make_pets && @pet_goal < 1
-      if @settings.pet_boxes_on_hand > 0
-        @pet_goal = @settings.pet_boxes_on_hand - get_boxes(@settings.picking_pet_box_source).count
-        exit unless @pet_goal > 0
-      else
-        echo 'You need to have a pet_boxes_on_hand setting to disarm pets without an entered count.'
-        exit
-      end
+    if @settings.waggle_sets['pick']
+      DRC.wait_for_script_to_complete('buff', ['pick'])
     end
+  end
 
-    @harvest_traps = !@make_pets && @settings.harvest_traps
-    @component_container = @settings.component_container
-    @stop_pick_on_mindlock = !@make_pets && @settings.stop_pick_on_mindlock
-    @never_pick_blind = @settings.never_pick_blind
-    @always_pick_blind = @settings.pick_blind || @settings.always_pick_blind
-    @loot_nouns = @settings.lootables
-    echo "Loot nouns: #{@loot_nouns}" if UserVars.lockpick_debug
+  def stop_buffs
+    drca_stop_buffs(@settings, 'pick') if DRStats.barbarian? || DRStats.thief?
+  end
 
-    @trash_nouns = get_data('items').trash_nouns
-    echo "Trash nouns: #{@trash_nouns}" if UserVars.lockpick_debug
-
-    messages = get_data('picking').picking
-    @pick_careful = messages['pick_careful']
-    @pick_quick = messages['pick_quick']
-    @pick_blind = messages['pick_blind']
-    @pick_retry = messages['pick_retry']
-    @disarm_failed = messages['disarm_failed']
-    @disarm_retry = messages['disarm_retry']
-    @disarm_identify_failed = messages['disarm_identify_failed']
-    @disarm_too_hard = messages['disarm_too_hard']
-    @disarm_careful = messages['disarm_careful']
-    @disarm_quick = messages['disarm_quick']
-    @disarm_normal = messages['disarm_normal']
-
-    @lockpick_costs = messages['lockpick_costs']
-    if args.refill
-      refill_ring
-      exit
-    end
+  # TODO - Move to DRCA
+  def drca_stop_buffs(settings, set_name)
+    return unless settings.waggle_sets[set_name]
 
-    Flags.add('disarm-more', 'not fully disarmed', 'not yet fully disarmed', 'still has more to torment you with')
-    open_container(@settings.picking_box_source)
-    open_container(@settings.picking_box_storage)
-    open_container(@settings.picking_pet_box_source)
-    open_container(@settings.component_container)
-  end
+    spells = settings.waggle_sets[set_name]
 
-  def open_container(name)
-    return unless name
+    if DRStats.barbarian?
+      stop_barb_abilities
+    elsif DRStats.thief?
+      stop_khris(spells)
+    else
+      release_spells(spells)
+    end
+  end
 
-    fput("open my #{name}")
+  # TODO - Move to DRCA
+  def stop_barb_abilities
+    DRC.bput('meditate stop')
```

Comment:
`bput` will need match strings to not hang after 15 seconds, and each meditation has its own "stop" messaging. I don't see them documented yet in base-spells. Those messages either need to be added so can be referenced here, or,  if we don't care what the output is, then we can use the wildcard regex pattern or use `fput`.
```suggestion
    DRC.bput('meditate stop', '.*')
```

Reply:
Yeah. I can put in the .* there. I'd have to go back to looking at what the old pick did to see how it handled it.